### ISSUE TITLE
Audiobookshelf crashes randomly?

### ISSUE THREAD

**Issue by colvdv** [mikiher/audiobookshelf-windows#8]
Every few days I find that audiobookshelf has crashed and when I hover over the tray icon it dissapears. I then have to manually relaunch it and it starts up the server again.

This seems to happen at random times. This last time it was when I clicked refresh on a library, it then popped up "socket disconnected" and I checked what was up and it had crashed. I've been experiencing this every few days it would seem. 

Could someone assist me to troubleshoot and get to the bottom of this? Thanks!

**Comment by mikiher** [comment]
Can you please attach the log file so I can see if there's something there?

You can find it at `%LOCALAPPDATA%\AudiobookselfTray\logfile.txt` (which usually translates to `C:\Users\<YourUser>\AppData\Local\AudiobookshelfTray\logfile.txt`)

**Comment by colvdv** [comment]
[logfile.txt](https://github.com/mikiher/audiobookshelf-windows/files/14365286/logfile.txt)

Thanks

**Comment by mikiher** [comment]
I can see one server crash happening a couple of days ago (on Feb 20) during a quick match attempt. This seems to be some uncaught database bug.  I will open a server bug for this.

Can you please also add the following log files for the library scans that ran just prior to the crash, so I can get a clearer picture of what caused it:
- C:\Users\D-Admin\AppData\Local\Audiobookshelf\metadata\logs\scans\2024-02-20_d1a72115-2bb0-4819-a30e-816bb864d43e.txt
- C:\Users\D-Admin\AppData\Local\Audiobookshelf\metadata\logs\scans\2024-02-20_536c2c57-6366-4037-83dd-8b557c8ccd50.txt
- C:\Users\D-Admin\AppData\Local\Audiobookshelf\metadata\logs\scans\2024-02-20_9bd64de4-abdc-4d2b-a0e9-4192b2781d8c.txt

**_Note_**: this is the only actual crash I can identify in the log. 
You said that this happened a number of times (the log contains data since 2024-02-04)? 
I do see a number of times in which the tray app was started without the server having been properly shut down prior to that. This can happen normally when you reboot your computer, and I believe that this is what happened.

**Comment by colvdv** [comment]
[2024-02-20_9bd64de4-abdc-4d2b-a0e9-4192b2781d8c.txt](https://github.com/mikiher/audiobookshelf-windows/files/14381852/2024-02-20_9bd64de4-abdc-4d2b-a0e9-4192b2781d8c.txt)
[2024-02-20_536c2c57-6366-4037-83dd-8b557c8ccd50.txt](https://github.com/mikiher/audiobookshelf-windows/files/14381853/2024-02-20_536c2c57-6366-4037-83dd-8b557c8ccd50.txt)
[2024-02-20_d1a72115-2bb0-4819-a30e-816bb864d43e.txt](https://github.com/mikiher/audiobookshelf-windows/files/14381855/2024-02-20_d1a72115-2bb0-4819-a30e-816bb864d43e.txt)

Here they are. Thanks again!

**Comment by mikiher** [comment]
I opened a server bug. I will continue to investigate and report on that bug.

If you can, can you also please answer the question above - is this the only crash you saw in the period covered by the log file, or have you seen more crashes? If you saw more crashes, I need to consider additional actions.

> Note: this is the only actual crash I can identify in the log.
You said that this happened a number of times (the log contains data since 2024-02-04)?
I do see a number of times in which the tray app was started without the server having been properly shut down prior to that. This can happen normally when you reboot your computer, and I believe that this is what happened.

**Comment by colvdv** [comment]
I had 2 instances where I went to open the app on my phone and it couldn't connect, so I went to check the server status and as soon as I hovered over the tray icon, it disappeared. No error messages or popups that I saw.
Opened up Audiobookshelf again and it didn't act like it had crashed. After the second (could have even been a third instance) is when I opened the issue.

I know the computer didn't restart in both scenarios because I have another server that was still running and I have to always manually start it after booting up.
